criu/util.c

```diff
@@ -470,6 +472,193 @@ int copy_file(int fd_in, int fd_out, size_t bytes)
 	return ret;
 }
 
+#define BUFSIZE	(4096)
+
+static int copy_chunk_from_file(int fd, int img, off_t off, size_t len)
+{
+	char *buf = NULL;
+	int ret;
+
+	if (opts.remote) {
+		buf = xmalloc(BUFSIZE);
+		if (!buf)
+			return -1;
+	}
+
+	while (len > 0) {
+		if (opts.remote) {
+			ret = pread(fd, buf, min_t(size_t, BUFSIZE, len), off);
+			if (ret <= 0) {
+				pr_perror("Can't read from ghost file");
```

Comment:
nits: I think that if we are generalizing this function by moving to `util.c` it would be great if we change error messages to more common without mentioning `ghost` files. What do you think about it? We could also add additional `pr_err` to calling code to make logs more informative if needed.

Reply:
I agree. I amended the commit.

I don't think we need an additional `pr_err` as the current error messages will immediately follow `pr_info("Dumping ghost file contents")` in the logs